### DISCUSSION TITLE
Could not connect to this Shlink server (using HTTPS)

### DISCUSSION THREAD

**Discussion by SmartPhoneLover** [shlinkio/discussions#129] [Help wanted] [answered]
### Shlink version

3.6.4

### PHP version

Docker

### How do you serve Shlink

Other (explain in summary)

### Database engine

MariaDB

### Database version

10.11.5

### Current behavior

When I try to add my Shlink server into my local Shlink Web Client app, it shows the following error in the web console:

`Cross-origin request blocked: Same-origin policy does not allow reading remote resources at https://<subdomain>.<base_domain>.xxx/rest/v3/health. (Reason: Unsuccessful CORS request). Status code: (null)`

Let's explain how I set up all the things...
**Shlink**
It's running under Docker, on my server. It's publicly accessible via Cloudflare Tunnels.
The docker-compose was created with the HTTPS enabled, also the domain and subdomains loads with HTTPS.
When I try to visit the service using its public address, I can see the 404 error page, and I can interact with it via cmd.

It's public domain is: https://links.my-site.com (just an example)

**Shlink Web Client**
It's running under Docker, on my server. But, it's not publicly accessible, but being able to connect only from LAN via HTTPS using a local certificate created with DNS Challenge. It's configured with Nginx Proxy Manager.

It's local domain: https://shlink-wc.my-apps.lan (just an example)

So, both Shlink and Shlink Web Client services are under HTTPS.

### Expected behavior

The Shlink Web Client instance should connect to the Shlink public instance.

### How to reproduce

Try to connect to it.

**Comment by acelaya** [comment]
The error means that the browser does not consider the server and client certificates equally secure.

By the client domain name, I'm going to assume you used a self-signed certificate. If that's the case, then that is the problem.

Browsers will allow typically insecure scenarios only when localhost is involved, so running the web client locally would work, even without https.

**Reply by SmartPhoneLover** [reply]
Thank you for answering.

But, I have another question related to it.
I also tried with 'https://app.shlink.io/', but I could connect the public Shlink to it neither.

Could it be something related to Cloudflare actually?

**Reply by acelaya** [reply, marked as answer]
Ah, I actually miss-read the error.

Yes, cloudflare is removing the CORS headers generated by Shlink. You have to find the way to tell cloudflare to preserve them.
